pkg/daemon/drain.go

```diff
@@ -254,13 +267,28 @@ func isSafeContainerRegistryConfChanges(oldIgnConfig, newIgnConfig ign3types.Con
 
 // searchRegistryMirror does lookup of a mirror in the mirroList specified for a registry
 // Returns true if found
-func searchRegistryMirror(loc string, mirrors []sysregistriesv2.Endpoint) bool {
+func searchRegistryMirror(loc string, mirrors []sysregistriesv2.Endpoint) (bool, sysregistriesv2.Endpoint) {
 	found := false
 	for _, m := range mirrors {
 		if m.Location == loc {
 			found = true
+			return found, m
+		}
+	}
+	return found, sysregistriesv2.Endpoint{}
+}
+
+func allDigestOnlyMirror(reg sysregistriesv2.Registry) bool {
+	tagAllowedMirrors := false
+	if len(reg.Mirrors) == 0 {
+		tagAllowedMirrors = true
```

Comment:
Why is this correct?

Reply:
replace this code with checking registry level MirrorByDigestOnly.